File: ios/expo/views/SmileIDExpoEnhancedDocumentVerificationView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of EnhancedDocumentVerification using the new architecture
 */
class SmileIDExpoEnhancedDocumentVerificationView: SmileIDExpoView {
    
    var countryCode: String?
    var documentType: String?
    var idAspectRatio: Double?
    var captureBothSides: Bool = true
    var bypassSelfieCaptureWithFile: String?
    var allowGalleryUpload: Bool = false
    var allowNewEnroll: Bool = false
    var useStrictMode: Bool = false
    var consentInformation: [String: Any?]?
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNEnhancedDocumentVerification(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.userId = userId
        config.jobId = jobId
        config.allowAgentMode = allowAgentMode
        config.allowNewEnroll = allowNewEnroll
        config.showInstructions = showInstructions
        config.skipApiSubmission = skipApiSubmission
        config.showAttribution = showAttribution
        config.useStrictMode = useStrictMode
        config.extraPartnerParams = extraPartnerParams
        config.countryCode = countryCode
        config.documentType = documentType
        if let aspectRatio = idAspectRatio {
            config.idAspectRatio = Float(aspectRatio)
        }
        config.captureBothSides = captureBothSides
        config.bypassSelfieCaptureWithFilePath = bypassSelfieCaptureWithFile
        config.allowGalleryUpload = allowGalleryUpload
        
        // Convert consentInformation dictionary to ConsentInfo object
        if let consentDict = consentInformation {
            config.consentInformation = parseConsentInfo(consentDict)
        }
        
        return config
    }
    
    private func parseConsentInfo(_ dict: [String: Any?]) -> ConsentInfo {
        return ConsentInfo(
            isVerifiedProcess: dict["isVerifiedProcess"] as? Bool ?? false,
            name: dict["name"] as? String ?? "",
            otherConsents: dict["otherConsents"] as? [String] ?? []
        )
    }
}

File: package/android/src/expo/com/smileidentity/expo/views/SmileIDExpoEnhancedDocumentVerificationView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNEnhancedDocumentVerification
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Expo implementation of EnhancedDocumentVerification using the new architecture
 */
class SmileIDExpoEnhancedDocumentVerificationView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var countryCode: String? = null
    var documentType: String? = null
    var allowGallerySelection: Boolean = false
    var captureBothSides: Boolean = true
    var consentInformation: Map<String, Any?>? = null

    override fun renderContent() {
        setContentWithTheme {
            RenderEnhancedDocumentVerificationContent()
        }
    }

    @Composable
    private fun RenderEnhancedDocumentVerificationContent() {
        val config = SmileIDViewConfig(
            userId = userId,
            jobId = jobId,
            allowAgentMode = allowAgentMode ?: false,
            showInstructions = showInstructions,
            skipApiSubmission = skipApiSubmission,
            showAttribution = showAttribution,
            extraPartnerParams = extraPartnerParams,
            countryCode = countryCode,
            documentType = documentType,
            allowGallerySelection = allowGallerySelection,
            captureBothSides = captureBothSides,
            consentInformation = consentInformation
        )

        SmileID.RNEnhancedDocumentVerification(
            config = config,
            onResult = { result ->
                handleResultCallback(result)
            }
        )
    }
}